src/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -52,6 +52,7 @@ public class SmtpClient : IDisposable
         private ContextAwareResult _operationCompletedResult = null;
         private AsyncOperation _asyncOp = null;
         private static AsyncCallback s_contextSafeCompleteCallback = new AsyncCallback(ContextSafeCompleteCallback);
+        private static NetworkCredential s_defaultNetworkCredential = CredentialCache.DefaultNetworkCredentials;
```

Comment:
> And CredentialCache is checking environment permissions etc

It's just returning a static field:
http://source.dot.net/#System.Net.Primitives/System/Net/CredentialCache.cs,226

> is caching harmful here?

It adds an unnecessary static field if the value being retrieved is already in a static field.